### PR TITLE
create packages and refolder classes; create useCases (GetMovieUseCas…

### PR DESCRIPTION
**Pull request by A20Karlikov**
…e and GetMovieById)

### CODE REVIEW

File: app/src/main/java/com/io/moviO/data/MoviesRepository.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package com.io.moviO.data
+
+import com.io.moviO.R
+
+object MoviesRepository {
+
+    private var moviesData = mutableListOf<Movie>()
+
+    fun getMovies(): List<Movie> {
+        setMovies()
```
Comment (stantt) [inline app/src/main/java/com/io/moviO/data/MoviesRepository.kt]: `setMovies()` can be in the init block

Comment (stantt) [reply app/src/main/java/com/io/moviO/data/MoviesRepository.kt]: A function should be responsible only for one operation. The operation it's named after. If the function is getMovies it doing a "setMovies" sounds wrong :)

File: app/src/main/java/com/io/moviO/data/MoviesRepository.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        return moviesData
+    }
+
+    private fun setMovies() {
+        moviesData = listOf(
```
Comment (stantt) [inline app/src/main/java/com/io/moviO/data/MoviesRepository.kt]: listOf should be mutableListOf. And remove the cast at the end

File: app/src/main/java/com/io/moviO/domain/GetMovieByIdUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.io.moviO.data.Movie
+import com.io.moviO.data.MoviesRepository
+
+class GetMovieByIdUseCase {
+
+    private val moviesRepo = MoviesRepository
+
+    fun getMovieById(id: String): Movie {
+        val movie = moviesRepo.getMovies().find { movie -> movie.id == id }
```
Comment (stantt) [inline app/src/main/java/com/io/moviO/domain/GetMovieByIdUseCase.kt]: not wrong to use `find`, but `first` would be a better fit.

File: app/src/main/java/com/io/moviO/domain/GetMovieByIdUseCase.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        return movie!!
```
Comment (stantt) [inline app/src/main/java/com/io/moviO/domain/GetMovieByIdUseCase.kt]: why the !! at the end?

File: app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt [hunk 5]
```diff
@@ -16,7 +18,8 @@ class MovieDetailsFragment : Fragment(R.layout.fragment_movie_details) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding = FragmentMovieDetailsBinding.bind(view)
-        requireArguments().getParcelable<Movie>(ARG_MOVIE)?.let {
+        requireArguments().getString(ARG_MOVIE)?.let { viewModel.setMovie(it) }
```
Comment (Golemanova) [inline app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt]: setMovie? Is that really what this method do?

Comment (stantt) [inline app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt]: setMovie sounds like the ViewModel has a property and you set it. :D 

I'd say - requestMovie or getMovieById? sounds more on point

File: app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt [hunk 6]
```diff
@@ -25,12 +28,12 @@ class MovieDetailsFragment : Fragment(R.layout.fragment_movie_details) {
...
             }
-        }
+        })
     }
 
     companion object {
-        fun newInstance(movie: Movie): MovieDetailsFragment = MovieDetailsFragment().also {
-            it.arguments = Bundle().apply { putParcelable(ARG_MOVIE, movie) }
+        fun newInstance(id: String): MovieDetailsFragment = MovieDetailsFragment().also {
+            it.arguments = Bundle().apply { putString(ARG_MOVIE, id) }
```
Comment (stantt) [inline app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt]: ARG_MOVIE could be ARG_MOVIE_ID

File: app/src/main/java/com/io/moviO/movieDetails/MovieDetailsViewModel.kt [hunk 7]
```diff
@@ -1,11 +1,19 @@
...
+import com.io.moviO.domain.GetMovieByIdUseCase
 
 class MovieDetailsViewModel : ViewModel() {
 
     private val _movie = MutableLiveData<Movie>()
     val movie: LiveData<Movie> = _movie
+
+    private val useCase = GetMovieByIdUseCase()
+
+    fun setMovie(id: String) {
```
Comment (stantt) [inline app/src/main/java/com/io/moviO/movieDetails/MovieDetailsViewModel.kt]: see - VM. setMovie calls UC. getMovieById 

You think the method sets things, while it actually gets things :D

**Review by stantt — APPROVED** [review]

File: app/src/main/java/com/io/moviO/data/MoviesRepository.kt (hunk 1 above)
Comment (A20Karlikov) [reply app/src/main/java/com/io/moviO/data/MoviesRepository.kt]: fixed

File: app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt (hunk 5 above)
Comment (A20Karlikov) [reply app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt]: fixed

File: app/src/main/java/com/io/moviO/movieDetails/MovieDetailsViewModel.kt (hunk 7 above)
Comment (A20Karlikov) [reply app/src/main/java/com/io/moviO/movieDetails/MovieDetailsViewModel.kt]: fixed

File: app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt (hunk 5 above)
Comment (A20Karlikov) [reply app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt]: fixed

File: app/src/main/java/com/io/moviO/domain/GetMovieByIdUseCase.kt (hunk 4 above)
Comment (A20Karlikov) [reply app/src/main/java/com/io/moviO/domain/GetMovieByIdUseCase.kt]: when use find instead of first, kotlin requires it. Now its gone :D

**Review by stantt — APPROVED** [review]

**Pull request merged**